cpp/src/topology/topology.cuh

```diff
@@ -45,25 +59,28 @@ namespace detail {
  * [col_indices[row_offsets[col_indx]]..col_indices[row_offsets[col_indx+1]])); return flag;
  *
  * @tparam IndexT type of indices for rows and columns
- * @tparam Vector type of the container used to hold buffers
- * @param d_row_offsets CSR row ofssets array
- * @param d_col_indices CSR column indices array
+ * @param handle raft handle
+ * @param nrows number of vertices
+ * @param ptr_r_o CSR row ofssets array
+ * @param nnz number of edges
+ * @param ptr_c_i CSR column indices array
  */
-template <typename IndexT, template <typename, typename...> typename Vector>
-bool check_symmetry(const Vector<IndexT>& d_row_offsets, const Vector<IndexT>& d_col_indices)
+template <typename IndexT>
+bool check_symmetry(raft::handle_t const& handle,
+                    IndexT nrows,
+                    const IndexT* ptr_r_o,
+                    IndexT nnz,
+                    const IndexT* ptr_c_i)
 {
-  auto nnz    = d_col_indices.size();
-  auto nrows  = d_row_offsets.size() - 1;
   using BoolT = bool;
-  Vector<BoolT> d_flags(nrows, 1);
+  rmm::device_uvector<BoolT> d_flags(nrows, handle.get_stream());
+  thrust::fill_n(rmm::exec_policy(handle.get_stream_view()), d_flags.begin(), nrows, true);
 
-  const IndexT* ptr_r_o = thrust::raw_pointer_cast(&d_row_offsets.front());
-  const IndexT* ptr_c_i = thrust::raw_pointer_cast(&d_col_indices.front());
-  BoolT* start_flags    = thrust::raw_pointer_cast(&d_flags.front());  // d_flags.begin();
-  BoolT* end_flags      = start_flags + nrows;
+  BoolT* start_flags = d_flags.data();  // d_flags.begin();
+  BoolT* end_flags   = start_flags + nrows;
   BoolT init{1};
   return thrust::transform_reduce(
-    thrust::device,
+    rmm::exec_policy(handle.get_stream_view()),
```

Comment:
So in the below, you're checking only in case ```if (col_indx > row_indx) {```.

Is this sufficient?

Say ```row_indx``` is 3. Will this cover the case ```row_indx, col_indx``` = (3, 2) exists but (2, 3) missing?

Reply:
So, this is an old function, to which I only changed the `exec` policy to make it compliant to `rmm`. I'd have to double check.